src/Slices/ServiceInventory/UI/Components/ForceStateAction.tsx

```diff
@@ -0,0 +1,242 @@
+  const handleModalToggle = () => {
+    setIsModalOpen(!isModalOpen);
+  };
+  const dropdownItems = targets.map((target) => (
+    <DropdownItem
+      key={target}
+      value={target}
+      data-testid={`${id}-${target}-expert`}
+    >
+      {target}
+    </DropdownItem>
+  ));
+  const { commandResolver, environmentModifier } =
+    useContext(DependencyContext);
+  const trigger = commandResolver.useGetTrigger<"TriggerForceState">({
+    kind: "TriggerForceState",
+    service_entity,
+    id,
+    version,
+  });
+  const isHalted = environmentModifier.useIsHalted();
+
+  const onSubmit = async (targetState: string) => {
+    const result = await trigger(targetState);
+    if (Maybe.isSome(result)) {
+      setStateErrorMessage(result.value);
+    }
+  };
+
+  const onSelect = (event) => {
+    setIsDropdownOpen(false);
+    setTargetState(event.target.text);
+    setConfirmationText(
+      words("inventory.statustab.forceState.message")(
+        instance_identity,
+        event.target.text
+      )
+    );
+    handleModalToggle();
+  };
+
+  return (
+    <>
+      {stateErrorMessage && (
+        <ToastAlertMessage
+          stateErrorMessage={stateErrorMessage}
+          id={id}
+          setStateErrorMessage={setStateErrorMessage}
+        />
+      )}
+      <ActionDisabledTooltip
+        ariaLabel={words("inventory.statustab.forceState")}
+        tooltipContent={
+          isHalted
+            ? words("environment.halt.tooltip")
+            : words("inventory.statustab.actionDisabled")
+        }
+      >
+        <StyledDropdown
+          toggle={
+            <DropdownToggle
+              data-testid={`${id}-force-state-toggle`}
+              onToggle={() => setIsDropdownOpen(!isDropdownOpen)}
+              toggleIndicator={CaretDownIcon}
+            >
+              {words("inventory.statustab.forceState")}
+            </DropdownToggle>
+          }
+          dropdownItems={dropdownItems}
+          isOpen={isDropdownOpen}
+          onSelect={onSelect}
+        />
+      </ActionDisabledTooltip>
+      <ConfirmationModal
+        confirmationText={confirmationText}
+        onSetInstanceState={onSubmit}
+        id={id}
+        targetState={targetState}
+        isModalOpen={isModalOpen}
+        setIsModalOpen={handleModalToggle}
+        setErrorMessage={setStateErrorMessage}
+      />
+    </>
+  );
+};
+
+interface ModalProps {
+  confirmationText: string;
+  id: string;
+  targetState: string;
+  onSetInstanceState: (targetState: string) => Promise<void>;
+  isModalOpen: boolean;
+  setIsModalOpen: (isOpen: boolean) => void;
+  setErrorMessage: (message: string) => void;
+}
+
+const ConfirmationModal: React.FC<ModalProps> = ({
+  confirmationText,
+  onSetInstanceState,
+  isModalOpen,
+  setIsModalOpen,
+  id,
+  targetState,
+}) => {
+  const handleModalToggle = () => {
+    setIsModalOpen(!isModalOpen);
+  };
+  const handleConfirm = async () => {
+    handleModalToggle();
```

Comment:
It's best practice to toggle the modal after whatever action it's supposed to execute.